dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/ClaimedTaskRestRepository.java

```diff
@@ -0,0 +1,163 @@
+
+    @Autowired
+    XmlWorkflowService workflowService;
+
+    @Autowired
+    WorkflowRequirementsService workflowRequirementsService;
+
+    @Override
+    public ClaimedTaskRest findOne(Context context, Integer id) {
+        ClaimedTask task = null;
+        try {
+            task = claimedTaskService.find(context, id);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        if (task == null) {
+            return null;
+        }
+        return converter.fromModel(task);
+    }
+
+    @SearchRestMethod(name = "findByUser")
+    public Page<ClaimedTaskRest> findByUser(@Parameter(value = "uuid") UUID userID, Pageable pageable) {
+        List<ClaimedTask> tasks = null;
+        try {
+            Context context = obtainContext();
+            EPerson ep = epersonService.find(context, userID);
+            tasks = claimedTaskService.findByEperson(context, ep);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        Page<ClaimedTaskRest> page = utils.getPage(tasks, pageable).map(converter);
+        return page;
+    }
+
+    @Override
+    public Class<ClaimedTaskRest> getDomainClass() {
+        return ClaimedTaskRest.class;
+    }
+
+    @Override
+    public ClaimedTaskResource wrapResource(ClaimedTaskRest task, String... rels) {
+        return new ClaimedTaskResource(task, utils, rels);
+    }
+
+    @Override
+    protected ClaimedTaskRest action(Context context, HttpServletRequest request, Integer id)
+        throws SQLException, IOException, AuthorizeException {
+        ClaimedTask task = null;
+        task = claimedTaskService.find(context, id);
+        XmlWorkflowServiceFactory factory = (XmlWorkflowServiceFactory) XmlWorkflowServiceFactory.getInstance();
+        Workflow workflow;
+        try {
+            workflow = factory.getWorkflowFactory().getWorkflow(task.getWorkflowItem().getCollection());
+
+            Step step = workflow.getStep(task.getStepID());
+            WorkflowActionConfig currentActionConfig = step.getActionConfig(task.getActionID());
+            workflowService
+                .doState(context, context.getCurrentUser(), request, task.getWorkflowItem().getID(), workflow,
+                    currentActionConfig);
+            if (!Action.getErrorFields(request).isEmpty()) {
+                throw new UnprocessableEntityException("Missing required fields");
+            }
+            // workflowRequirementsService.removeClaimedUser(context, task.getWorkflowItem(), task.getOwner(), task
+            // .getStepID());
+            context.addEvent(new Event(Event.MODIFY, Constants.ITEM, task.getWorkflowItem().getItem().getID(), null,
+                itemService.getIdentifiers(context, task.getWorkflowItem().getItem())));
+        } catch (WorkflowConfigurationException | MessagingException | WorkflowException e) {
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
We should return a more useful error here. Something like "Could not claim task with ID" + id.

Reply:
see comment above. I have deal with the workflow exception in a more direct/useful way

Reply:
Can you please add the id in the error message as suggested by @tdonohue above

Reply:
As explained in this other comment https://github.com/DSpace/DSpace/pull/2312#discussion_r259634147
I will prefer to not do that. This will be the first place where we start to replicate such information and I'm not convinced that it is necessary, so I will prefer to take an uniform decision about that in DS-3992
Please note that it is very easy to add a generic uniform logging of all the exception that will show the verb, endpoint and parameters of the original call (and you also get this mostly out of box in the access log of apache or tomcat). The REST client already now the information that you are asking to add, in a structural way, because it knows the hit endpoint and verb.

Anyway, if both @tdonohue and @benbosman prefer to have this message in place I will add it

Reply:
@abollini : That's fine by me. You are correct that the client should already know which ID it was trying to perform the request on anyways. So, I'll withdraw my request here.  I'd still recommend we look at implementing DS-3992 soon though -- it doesn't seem like something we should leave till the "last minute".